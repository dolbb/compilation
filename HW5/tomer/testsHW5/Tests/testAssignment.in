int Int(){return 500;}
byte Byte(){return 3b;}
void main(){
int i1 = 400;
byte c1 = 2b;
int t1;
t1 = 300;
printi(t1);
print("\n");
t1 = i1;
printi(t1);
print("\n");
t1 = Int();
printi(t1);
print("\n");
t1 = (200+400);
printi(t1);
print("\n");
t1 = 350*2;
printi(t1);
print("\n");
t1 = (300);
printi(t1);
print("\n");
t1 = (i1);
printi(t1);
print("\n");
t1 = (Int());
printi(t1);
print("\n");
t1 = ((200+400));
printi(t1);
print("\n");
t1 = (350*2);
printi(t1);
print("\n");
t1 = 300 + 300;
printi(t1);
print("\n");
t1 = 300 + i1;
printi(t1);
print("\n");
t1 = 300 + Int();
printi(t1);
print("\n");
t1 = 300 + (200+400);
printi(t1);
print("\n");
t1 = 300 + 350*2;
printi(t1);
print("\n");
t1 = i1 + 300;
printi(t1);
print("\n");
t1 = i1 + i1;
printi(t1);
print("\n");
t1 = i1 + Int();
printi(t1);
print("\n");
t1 = i1 + (200+400);
printi(t1);
print("\n");
t1 = i1 + 350*2;
printi(t1);
print("\n");
t1 = Int() + 300;
printi(t1);
print("\n");
t1 = Int() + i1;
printi(t1);
print("\n");
t1 = Int() + Int();
printi(t1);
print("\n");
t1 = Int() + (200+400);
printi(t1);
print("\n");
t1 = Int() + 350*2;
printi(t1);
print("\n");
t1 = (200+400) + 300;
printi(t1);
print("\n");
t1 = (200+400) + i1;
printi(t1);
print("\n");
t1 = (200+400) + Int();
printi(t1);
print("\n");
t1 = (200+400) + (200+400);
printi(t1);
print("\n");
t1 = (200+400) + 350*2;
printi(t1);
print("\n");
t1 = 350*2 + 300;
printi(t1);
print("\n");
t1 = 350*2 + i1;
printi(t1);
print("\n");
t1 = 350*2 + Int();
printi(t1);
print("\n");
t1 = 350*2 + (200+400);
printi(t1);
print("\n");
t1 = 350*2 + 350*2;
printi(t1);
print("\n");
t1 = 300 * 300;
printi(t1);
print("\n");
t1 = 300 * i1;
printi(t1);
print("\n");
t1 = 300 * Int();
printi(t1);
print("\n");
t1 = 300 * (200+400);
printi(t1);
print("\n");
t1 = 300 * 350*2;
printi(t1);
print("\n");
t1 = i1 * 300;
printi(t1);
print("\n");
t1 = i1 * i1;
printi(t1);
print("\n");
t1 = i1 * Int();
printi(t1);
print("\n");
t1 = i1 * (200+400);
printi(t1);
print("\n");
t1 = i1 * 350*2;
printi(t1);
print("\n");
t1 = Int() * 300;
printi(t1);
print("\n");
t1 = Int() * i1;
printi(t1);
print("\n");
t1 = Int() * Int();
printi(t1);
print("\n");
t1 = Int() * (200+400);
printi(t1);
print("\n");
t1 = Int() * 350*2;
printi(t1);
print("\n");
t1 = (200+400) * 300;
printi(t1);
print("\n");
t1 = (200+400) * i1;
printi(t1);
print("\n");
t1 = (200+400) * Int();
printi(t1);
print("\n");
t1 = (200+400) * (200+400);
printi(t1);
print("\n");
t1 = (200+400) * 350*2;
printi(t1);
print("\n");
t1 = 350*2 * 300;
printi(t1);
print("\n");
t1 = 350*2 * i1;
printi(t1);
print("\n");
t1 = 350*2 * Int();
printi(t1);
print("\n");
t1 = 350*2 * (200+400);
printi(t1);
print("\n");
t1 = 350*2 * 350*2;
printi(t1);
print("\n");
t1 = 1b + 300;
printi(t1);
print("\n");
t1 = 1b + i1;
printi(t1);
print("\n");
t1 = 1b + Int();
printi(t1);
print("\n");
t1 = 1b + (200+400);
printi(t1);
print("\n");
t1 = 1b + 350*2;
printi(t1);
print("\n");
t1 = c1 + 300;
printi(t1);
print("\n");
t1 = c1 + i1;
printi(t1);
print("\n");
t1 = c1 + Int();
printi(t1);
print("\n");
t1 = c1 + (200+400);
printi(t1);
print("\n");
t1 = c1 + 350*2;
printi(t1);
print("\n");
t1 = Byte() + 300;
printi(t1);
print("\n");
t1 = Byte() + i1;
printi(t1);
print("\n");
t1 = Byte() + Int();
printi(t1);
print("\n");
t1 = Byte() + (200+400);
printi(t1);
print("\n");
t1 = Byte() + 350*2;
printi(t1);
print("\n");
t1 = (1b+3b) + 300;
printi(t1);
print("\n");
t1 = (1b+3b) + i1;
printi(t1);
print("\n");
t1 = (1b+3b) + Int();
printi(t1);
print("\n");
t1 = (1b+3b) + (200+400);
printi(t1);
print("\n");
t1 = (1b+3b) + 350*2;
printi(t1);
print("\n");
t1 = 2b*4b + 300;
printi(t1);
print("\n");
t1 = 2b*4b + i1;
printi(t1);
print("\n");
t1 = 2b*4b + Int();
printi(t1);
print("\n");
t1 = 2b*4b + (200+400);
printi(t1);
print("\n");
t1 = 2b*4b + 350*2;
printi(t1);
print("\n");
t1 = 300 + 1b;
printi(t1);
print("\n");
t1 = 300 + c1;
printi(t1);
print("\n");
t1 = 300 + Byte();
printi(t1);
print("\n");
t1 = 300 + (1b+3b);
printi(t1);
print("\n");
t1 = 300 + 2b*4b;
printi(t1);
print("\n");
t1 = i1 + 1b;
printi(t1);
print("\n");
t1 = i1 + c1;
printi(t1);
print("\n");
t1 = i1 + Byte();
printi(t1);
print("\n");
t1 = i1 + (1b+3b);
printi(t1);
print("\n");
t1 = i1 + 2b*4b;
printi(t1);
print("\n");
t1 = Int() + 1b;
printi(t1);
print("\n");
t1 = Int() + c1;
printi(t1);
print("\n");
t1 = Int() + Byte();
printi(t1);
print("\n");
t1 = Int() + (1b+3b);
printi(t1);
print("\n");
t1 = Int() + 2b*4b;
printi(t1);
print("\n");
t1 = (200+400) + 1b;
printi(t1);
print("\n");
t1 = (200+400) + c1;
printi(t1);
print("\n");
t1 = (200+400) + Byte();
printi(t1);
print("\n");
t1 = (200+400) + (1b+3b);
printi(t1);
print("\n");
t1 = (200+400) + 2b*4b;
printi(t1);
print("\n");
t1 = 350*2 + 1b;
printi(t1);
print("\n");
t1 = 350*2 + c1;
printi(t1);
print("\n");
t1 = 350*2 + Byte();
printi(t1);
print("\n");
t1 = 350*2 + (1b+3b);
printi(t1);
print("\n");
t1 = 350*2 + 2b*4b;
printi(t1);
print("\n");
t1 = 1b * 300;
printi(t1);
print("\n");
t1 = 1b * i1;
printi(t1);
print("\n");
t1 = 1b * Int();
printi(t1);
print("\n");
t1 = 1b * (200+400);
printi(t1);
print("\n");
t1 = 1b * 350*2;
printi(t1);
print("\n");
t1 = c1 * 300;
printi(t1);
print("\n");
t1 = c1 * i1;
printi(t1);
print("\n");
t1 = c1 * Int();
printi(t1);
print("\n");
t1 = c1 * (200+400);
printi(t1);
print("\n");
t1 = c1 * 350*2;
printi(t1);
print("\n");
t1 = Byte() * 300;
printi(t1);
print("\n");
t1 = Byte() * i1;
printi(t1);
print("\n");
t1 = Byte() * Int();
printi(t1);
print("\n");
t1 = Byte() * (200+400);
printi(t1);
print("\n");
t1 = Byte() * 350*2;
printi(t1);
print("\n");
t1 = (1b+3b) * 300;
printi(t1);
print("\n");
t1 = (1b+3b) * i1;
printi(t1);
print("\n");
t1 = (1b+3b) * Int();
printi(t1);
print("\n");
t1 = (1b+3b) * (200+400);
printi(t1);
print("\n");
t1 = (1b+3b) * 350*2;
printi(t1);
print("\n");
t1 = 2b*4b * 300;
printi(t1);
print("\n");
t1 = 2b*4b * i1;
printi(t1);
print("\n");
t1 = 2b*4b * Int();
printi(t1);
print("\n");
t1 = 2b*4b * (200+400);
printi(t1);
print("\n");
t1 = 2b*4b * 350*2;
printi(t1);
print("\n");
t1 = 300 * 1b;
printi(t1);
print("\n");
t1 = 300 * c1;
printi(t1);
print("\n");
t1 = 300 * Byte();
printi(t1);
print("\n");
t1 = 300 * (1b+3b);
printi(t1);
print("\n");
t1 = 300 * 2b*4b;
printi(t1);
print("\n");
t1 = i1 * 1b;
printi(t1);
print("\n");
t1 = i1 * c1;
printi(t1);
print("\n");
t1 = i1 * Byte();
printi(t1);
print("\n");
t1 = i1 * (1b+3b);
printi(t1);
print("\n");
t1 = i1 * 2b*4b;
printi(t1);
print("\n");
t1 = Int() * 1b;
printi(t1);
print("\n");
t1 = Int() * c1;
printi(t1);
print("\n");
t1 = Int() * Byte();
printi(t1);
print("\n");
t1 = Int() * (1b+3b);
printi(t1);
print("\n");
t1 = Int() * 2b*4b;
printi(t1);
print("\n");
t1 = (200+400) * 1b;
printi(t1);
print("\n");
t1 = (200+400) * c1;
printi(t1);
print("\n");
t1 = (200+400) * Byte();
printi(t1);
print("\n");
t1 = (200+400) * (1b+3b);
printi(t1);
print("\n");
t1 = (200+400) * 2b*4b;
printi(t1);
print("\n");
t1 = 350*2 * 1b;
printi(t1);
print("\n");
t1 = 350*2 * c1;
printi(t1);
print("\n");
t1 = 350*2 * Byte();
printi(t1);
print("\n");
t1 = 350*2 * (1b+3b);
printi(t1);
print("\n");
t1 = 350*2 * 2b*4b;
printi(t1);
print("\n");
byte t2;
t2 = 1b;
printi(t2);
print("\n");
t2 = c1;
printi(t2);
print("\n");
t2 = Byte();
printi(t2);
print("\n");
t2 = (1b+3b);
printi(t2);
print("\n");
t2 = 2b*4b;
printi(t2);
print("\n");
t2 = (1b);
printi(t2);
print("\n");
t2 = (c1);
printi(t2);
print("\n");
t2 = (Byte());
printi(t2);
print("\n");
t2 = ((1b+3b));
printi(t2);
print("\n");
t2 = (2b*4b);
printi(t2);
print("\n");
t2 = 1b * 1b;
printi(t2);
print("\n");
t2 = 1b * c1;
printi(t2);
print("\n");
t2 = 1b * Byte();
printi(t2);
print("\n");
t2 = 1b * (1b+3b);
printi(t2);
print("\n");
t2 = 1b * 2b*4b;
printi(t2);
print("\n");
t2 = c1 * 1b;
printi(t2);
print("\n");
t2 = c1 * c1;
printi(t2);
print("\n");
t2 = c1 * Byte();
printi(t2);
print("\n");
t2 = c1 * (1b+3b);
printi(t2);
print("\n");
t2 = c1 * 2b*4b;
printi(t2);
print("\n");
t2 = Byte() * 1b;
printi(t2);
print("\n");
t2 = Byte() * c1;
printi(t2);
print("\n");
t2 = Byte() * Byte();
printi(t2);
print("\n");
t2 = Byte() * (1b+3b);
printi(t2);
print("\n");
t2 = Byte() * 2b*4b;
printi(t2);
print("\n");
t2 = (1b+3b) * 1b;
printi(t2);
print("\n");
t2 = (1b+3b) * c1;
printi(t2);
print("\n");
t2 = (1b+3b) * Byte();
printi(t2);
print("\n");
t2 = (1b+3b) * (1b+3b);
printi(t2);
print("\n");
t2 = (1b+3b) * 2b*4b;
printi(t2);
print("\n");
t2 = 2b*4b * 1b;
printi(t2);
print("\n");
t2 = 2b*4b * c1;
printi(t2);
print("\n");
t2 = 2b*4b * Byte();
printi(t2);
print("\n");
t2 = 2b*4b * (1b+3b);
printi(t2);
print("\n");
t2 = 2b*4b * 2b*4b;
printi(t2);
print("\n");
t1 = 1b;
printi(t1);
print("\n");
t1 = c1;
printi(t1);
print("\n");
t1 = Byte();
printi(t1);
print("\n");
t1 = (1b+3b);
printi(t1);
print("\n");
t1 = 2b*4b;
printi(t1);
print("\n");
t1 = (1b);
printi(t1);
print("\n");
t1 = (c1);
printi(t1);
print("\n");
t1 = (Byte());
printi(t1);
print("\n");
t1 = ((1b+3b));
printi(t1);
print("\n");
t1 = (2b*4b);
printi(t1);
print("\n");
t1 = 1b * 1b;
printi(t1);
print("\n");
t1 = 1b * c1;
printi(t1);
print("\n");
t1 = 1b * Byte();
printi(t1);
print("\n");
t1 = 1b * (1b+3b);
printi(t1);
print("\n");
t1 = 1b * 2b*4b;
printi(t1);
print("\n");
t1 = c1 * 1b;
printi(t1);
print("\n");
t1 = c1 * c1;
printi(t1);
print("\n");
t1 = c1 * Byte();
printi(t1);
print("\n");
t1 = c1 * (1b+3b);
printi(t1);
print("\n");
t1 = c1 * 2b*4b;
printi(t1);
print("\n");
t1 = Byte() * 1b;
printi(t1);
print("\n");
t1 = Byte() * c1;
printi(t1);
print("\n");
t1 = Byte() * Byte();
printi(t1);
print("\n");
t1 = Byte() * (1b+3b);
printi(t1);
print("\n");
t1 = Byte() * 2b*4b;
printi(t1);
print("\n");
t1 = (1b+3b) * 1b;
printi(t1);
print("\n");
t1 = (1b+3b) * c1;
printi(t1);
print("\n");
t1 = (1b+3b) * Byte();
printi(t1);
print("\n");
t1 = (1b+3b) * (1b+3b);
printi(t1);
print("\n");
t1 = (1b+3b) * 2b*4b;
printi(t1);
print("\n");
t1 = 2b*4b * 1b;
printi(t1);
print("\n");
t1 = 2b*4b * c1;
printi(t1);
print("\n");
t1 = 2b*4b * Byte();
printi(t1);
print("\n");
t1 = 2b*4b * (1b+3b);
printi(t1);
print("\n");
t1 = 2b*4b * 2b*4b;
printi(t1);
print("\n");
}
