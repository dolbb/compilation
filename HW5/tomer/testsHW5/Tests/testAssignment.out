Loaded: ./exceptions.s
300
400
500
600
700
300
400
500
600
700
600
700
800
900
1000
700
800
900
1000
1100
800
900
1000
1100
1200
900
1000
1100
1200
1300
1000
1100
1200
1300
1400
90000
120000
150000
180000
210000
120000
160000
200000
240000
280000
150000
200000
250000
300000
350000
180000
240000
300000
360000
420000
210000
280000
350000
420000
490000
301
401
501
601
701
302
402
502
602
702
303
403
503
603
703
304
404
504
604
704
308
408
508
608
708
301
302
303
304
308
401
402
403
404
408
501
502
503
504
508
601
602
603
604
608
701
702
703
704
708
300
400
500
600
700
600
800
1000
1200
1400
900
1200
1500
1800
2100
1200
1600
2000
2400
2800
2400
3200
4000
4800
5600
300
600
900
1200
2400
400
800
1200
1600
3200
500
1000
1500
2000
4000
600
1200
1800
2400
4800
700
1400
2100
2800
5600
1
2
3
4
8
1
2
3
4
8
1
2
3
4
8
2
4
6
8
16
3
6
9
12
24
4
8
12
16
32
8
16
24
32
64
1
2
3
4
8
1
2
3
4
8
1
2
3
4
8
2
4
6
8
16
3
6
9
12
24
4
8
12
16
32
8
16
24
32
64
